src/test.js

```diff
@@ -389,7 +389,23 @@ Test.prototype = {
 			return true;
 		}
 
-		include = filter.charAt( 0 ) !== "!";
+		return regexFilter ?
+			this.regexFilter( regexFilter[1], regexFilter[2], regexFilter[3], fullName ) :
+			this.stringFilter( filter, fullName );
+	},
+
+	regexFilter: function( exclude, pattern, flags, fullName ) {
+		var regex = new RegExp( pattern, flags );
+		var include = regex.test( fullName );
+
+		return exclude ? !include : include;
```

Comment:
This `include` variable is dependent upon `fullName`, unlike the `include` variable in `stringFilter`. But honestly, I don't think it improves clarity anyway. What about something like
```js
var match = regex.test( fullName );
return match === !exclude;
```
?

`stringFilter` can be similarly cleaned up in another PR.